File: LV1/1.6/differentdistances/differentdistances.rb
loop do
  line = gets.split

  break if line.size == 1

  x1, y1, x2, y2, p = line.map(&:to_f)
  ans = ((x1 - x2).abs**p + (y1 - y2).abs**p)**(1 / p)
  puts ans
end

File: LV1/1.6/differentdistances/differentdistances.py
while True:
    line = input().split(' ')

    if len(line) == 1:
        break

    x1, y1, x2, y2, p = [float(i) for i in line]
    ans = (abs(x1-x2)**p+abs(y1-y2)**p)**(1/p)
    print(ans)
